### PR TITLE
fixed broken links in dev docs

### PR DESCRIPTION
**Pull request by clwgg**
## Description

I came upon two broken links in the dev docs -- this PR fixes them.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -180,10 +180,10 @@ skip to {ref}`sec_development_workflow_anothers_commit`.
    a local copy as detailed in {ref}`sec_development_getting_started_environment`.
 
 2. Make sure that your local repository has been configured with an
-   [upstream remote](https://help.github
-   .com/articles/configuring-a-remote-for-a-fork/):
+   [upstream remote](
+   https://help.github.com/articles/configuring-a-remote-for-a-fork/):
    ```bash
-   $ git remote add upstream git@github.com:tskit-dev/tskit.git
```
Comment (jeromekelleher) [inline docs/development.md:186]: I'm not sure about this change - I tend to use ssh access, but maybe most people don't?

I think the correct http link is ``https://github.com/tskit-dev/tskit.git`` in any case though.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, thanks @clwgg! One query about the http URL (and maybe it's wrong?)

File: docs/development.md (hunk 1 above)
Comment (grahamgower) [reply docs/development.md:186]: For the upstream, most folks wont have authenticated access and so must use the non-authenticated https url.

Comment (jeromekelleher) [reply docs/development.md:186]: Ah - thanks @grahamgower . OK, we just need to change to ``https://github.com/tskit-dev/tskit.git`` then

Comment (clwgg) [reply docs/development.md:186]: yeah, exactly! it works with or without the `.git` extension, but I'm happy to add it back in.

Comment (clwgg) [reply docs/development.md:186]: done!

**Review by benjeffery — APPROVED** [review]
LGTM! Thanks @clwgg!

**Pull request merged**